### ISSUE TITLE
GeoPackage and ArcGIS / GeoPackage and spatial indexes

### ISSUE THREAD

**Issue by heidivanparys** [ShapeChange/ShapeChange#263]
A GeoPackage file created with ShapeChange - and thus with https://github.com/ngageoint/geopackage-java - cannot be read by ArcGIS Pro (version used: 2.5.0). This seems to be an ArcGIS issue, however, it may be useful to know for others that are using this target.

I have a case at ESRI, see [this file](https://github.com/ShapeChange/ShapeChange/files/5437730/ArcGIS.GeoPackage.pdf) for a description of the issues that I encountered.

A workaround for now is to add the spatial indexes afterwards using e.g. [Spatialite 5.0.0](http://www.gaia-gis.it/gaia-sins/).

```sql
select autogpkgstart();
CREATE TABLE gpkg_extensions ( table_name TEXT, column_name TEXT, extension_name TEXT NOT NULL, definition TEXT NOT NULL, scope TEXT NOT NULL, CONSTRAINT ge_tce UNIQUE (table_name, column_name, extension_name) ); -- if this table is not yet present
select gpkgaddspatialindex('myfeaturetype','location');
-- at this point, data can be inserted: insert into myfeaturetype (label, location) values [...];
select autogpkgstop();
```

However, at some point, support for creating spatial indexes should be added to ShapeChange.

**Comment by heidivanparys** [comment]
Update: this should be fixed in ArcGIS Pro 2.8: https://support.esri.com/en/bugs/nimbus/QlVHLTAwMDEzNTQyNw==

**Comment by jechterhoff** [comment]
@heidivanparys Closing this issue now, since it appears to be resolved with ArcGIS Pro 2.8. The information (on the problem, workaround, and fix) will still be available to everyone as part of the closed issues.
